packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -460,6 +481,32 @@ export class AmqpConnection {
     originalHandlerName = 'unknown',
     consumeOptions?: ConsumeOptions
   ): Promise<ConsumerTag> {
+    if (msgOptions.batchOptions) {
```

Comment:
Based on my point, if you had a special symbol you wouldn't need to conditionally register as the Decorator would streamline that for you.